packages/signers/src/signable-message.ts

```diff
@@ -0,0 +1,20 @@
+import { TextEncoder } from 'text-encoding-impl';
+
+import { SignatureDictionary } from './types';
+
+/** Defines a message that needs signing and its current set of signatures if any. */
+export type SignableMessage = Readonly<{
+    content: Uint8Array;
+    signatures: SignatureDictionary;
+}>;
+
+/**
+ * Creates a signable message from a provided content.
+ * If a string is provided, it will be UTF-8 encoded.
+ */
+export function createSignableMessage(content: string | Uint8Array, signatures: SignatureDictionary = {}) {
+    return Object.freeze({
+        content: typeof content === 'string' ? new TextEncoder().encode(content) : content,
+        signatures: Object.freeze(signatures),
```

Comment:
This could be upsetting for the person who passed this object into this function. They may wonder why it's all of a sudden frozen.

Reply:
Moving this conversation [here](https://github.com/solana-labs/solana-web3.js/pull/1710#discussion_r1390419403).